From 9082fa57f3352f671578053f98884f0909a1cf84 Mon Sep 17 00:00:00 2001
From: Yuanjun Huang <yuanjun.huang@intel.com>
Date: Fri, 8 Jun 2018 15:42:05 +0800
Subject: [PATCH] Add option to disable video vpp if va driver is not ready.

For certain reason on some platform the va driver is not enabled,
then video vpp path should be disabled.

Change-Id: I4e64778e30b6552cba3cc4071d228239a5abff87
Test: video playback normal and no other side effect.
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 common/Android.mk                      | 5 +++++
 common/compositor/factory.cpp          | 4 ++++
 common/display/displayplanemanager.cpp | 3 ++-
 common/display/displayqueue.cpp        | 2 ++
 4 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/common/Android.mk b/common/Android.mk
index 30bef8f..2c6ac8c 100644
--- a/common/Android.mk
+++ b/common/Android.mk
@@ -62,6 +62,11 @@ LOCAL_CPPFLAGS += \
 	-DDISABLE_CURSOR_PLANE
 endif
 
+ifeq ($(strip $(HWC_DISABLE_VA_DRIVER)), true)
+LOCAL_CPPFLAGS += \
+	-DDISABLE_VA
+endif
+
 #LOCAL_CPPFLAGS += \
 #	-DENABLE_DOWNSCALING
 
diff --git a/common/compositor/factory.cpp b/common/compositor/factory.cpp
index 01714e2..950d8ec 100644
--- a/common/compositor/factory.cpp
+++ b/common/compositor/factory.cpp
@@ -61,7 +61,11 @@ Renderer* CreateMediaRenderer() {
 #ifdef USE_DC
   return NULL;
 #else
+#ifndef DISABLE_VA
   return new VARenderer();
+#else
+  return NULL;
+#endif
 #endif
 }
 
diff --git a/common/display/displayplanemanager.cpp b/common/display/displayplanemanager.cpp
index 79364ba..9583435 100644
--- a/common/display/displayplanemanager.cpp
+++ b/common/display/displayplanemanager.cpp
@@ -211,10 +211,11 @@ bool DisplayPlaneManager::ValidateLayers(
 #endif
           plane->SetInUse(true);
           DisplayPlaneState &last_plane = composition.back();
+#ifndef DISABLE_VA
           if (layer->IsVideoLayer()) {
             last_plane.SetVideoPlane(true);
           }
-
+#endif
           if (fall_back) {
             if (!validate_final_layers)
               validate_final_layers = !(last_plane.GetOffScreenTarget());
diff --git a/common/display/displayqueue.cpp b/common/display/displayqueue.cpp
index 0a131d4..798fb74 100644
--- a/common/display/displayqueue.cpp
+++ b/common/display/displayqueue.cpp
@@ -347,7 +347,9 @@ void DisplayQueue::GetCachedLayers(const std::vector<OverlayLayer>& layers,
                    !target_plane.IsVideoPlane()) {
           const OverlayLayer& layer = layers.at(source_layers.at(0));
           if (layer.IsVideoLayer()) {
+#ifndef DISABLE_VA
             target_plane.SetVideoPlane(true);
+#endif
             display_plane_manager_->MarkSurfacesForRecycling(
                 &target_plane, surfaces_not_inuse_, true);
             refresh_surfaces = true;
-- 
1.9.1

